### ISSUE TITLE
Slow to generate docs

### ISSUE THREAD

**Issue by rmundel** [knuckleswtf/scribe#120]
The verbose options aren't give me any insights (everything runs without errors) on why my documentation is taking very long to generate.
I have only 3 api routes and is taking like 10 minutos or so.
Any tips on where to start debugging it?

**Comment by shalvah** [comment]
What version are you on? Try v2 beta and see if there are any improvements.

I have speed improvements planned for v3, but that won't be until next year. In the meantime, you can also try disabling strategies you don't need by commenting them out in the config file. For instance, if you don't use Fractal Transformers, you can comment out the transformer tags strategy.

**Comment by rmundel** [comment]
I'm on 1.9.1. Just tried disabling every strategy there is, and the time was the same. Gonna to try v2.

**Comment by shalvah** [comment]
Also, are you limiting your config to your API routes only? Sometimes people forget.

**Comment by tjep82** [comment]
Had the same issue, i commented the following in the config strategies:

```
'bodyParameters' => [
            //\Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromFormRequest::class,
           \Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromBodyParamTag::class,
        ],
        'responses' => [
            //\Knuckles\Scribe\Extracting\Strategies\Responses\UseTransformerTags::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseTag::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseFileTag::class,
           //\Knuckles\Scribe\Extracting\Strategies\Responses\UseApiResourceTags::class,
           // \Knuckles\Scribe\Extracting\Strategies\Responses\ResponseCalls::class,
```

These are the strategies that is causing the slow process try that.

**Comment by R0ad-Runn3r** [comment]
Hey, I had the same issue but just inside my docker. Commenting these lines, @tjep82 suggested, "solved" it, but i don't think that this is the solution. I am actually not able to provide a solution, do you have any ideas?

**Comment by aaronflorey** [comment]
I've had the same issue and found it's because of a change that was made recently, that would create a transaction for every single database in config/database.php 

We're using the default database.php config file which includes mssql, pgsql, mysql and sqlite - so this package is trying to connect to every database and start a transaction to that database.

Even worse is that i don't have drivers for mssql or pgsql, so it was causing a "pdo driver not found" error, and then the other errors were "can't connect to server"

I'm not sure why this was added? but it was poorly done. Hopefully it's fixed soon.

**Comment by shalvah** [comment]
> I'm not sure why this was added? 

It was added because folks may use multiple database connections in their app, and would assume that response calls will be run in transactions for *all* of them. Any suggestions?

**Comment by matyhaty** [comment]
> Had the same issue, i commented the following in the config strategies:
> 
> ```
> 'bodyParameters' => [
>             //\Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromFormRequest::class,
>            \Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromBodyParamTag::class,
>         ],
>         'responses' => [
>             //\Knuckles\Scribe\Extracting\Strategies\Responses\UseTransformerTags::class,
>             \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseTag::class,
>             \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseFileTag::class,
>            //\Knuckles\Scribe\Extracting\Strategies\Responses\UseApiResourceTags::class,
>            // \Knuckles\Scribe\Extracting\Strategies\Responses\ResponseCalls::class,
> ```
> 
> These are the strategies that is causing the slow process try that.

This also worked for us

**Comment by shalvah** [comment]
> > I'm not sure why this was added?
> 
> It was added because folks may use multiple database connections in their app, and would assume that response calls will be run in transactions for _all_ of them. Any suggestions?

Best thing I can think of for this would be to wrap that in a timeout, to prevent things from getting out of hand. Nice investigation, btw.

**Comment by shalvah** [comment]
Hey, guys, I just released 2.2.0, which should fix this. in my tests, Scribe only takes about 3 seconds where it used to take 4 minutes.😄

Try it out and let me know if it works.
